File: backend/internal/models/session/session.go
package session

type Event string

const (
	DeviceConnected    Event = "DEVICE_CONNECTED"
	DeviceDisconnected Event = "DEVICE_DISCONNECTED"
)

type Session struct {
	SessionID string `json:"sessionId"`
	UserID    int64  `json:"userId"`
	Event     Event  `json:"sessionEvent"`
}

File: flink/inbound-messages-stream/src/main/java/com/universe/flink/inbound/models/Session.java
package com.universe.flink.inbound.models;

import com.fasterxml.jackson.annotation.JsonProperty;

public class Session {
    @JsonProperty("sessionId")
    private String sessionId;

    @JsonProperty("userId")
    private Long userId;

    @JsonProperty("sessionEvent")
    private SessionEvent sessionEvent;

    public String getSessionId() {
        return sessionId;
    }

    public Long getUserId() {
        return userId;
    }

    public SessionEvent getSessionEvent() {
        return sessionEvent;
    }
}